File: app/src/main/java/com/commit451/gitlab/adapter/GroupPagerAdapter.java
package com.commit451.gitlab.adapter;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.commit451.gitlab.R;
import com.commit451.gitlab.fragment.FeedFragment;
import com.commit451.gitlab.fragment.GroupMembersFragment;
import com.commit451.gitlab.fragment.ProjectsFragment;
import com.commit451.gitlab.model.api.Group;

import java.util.HashSet;
import java.util.Set;

/**
 * Group pager adapter
 */
public class GroupPagerAdapter extends FragmentPagerAdapter {

    private static final int SECTION_COUNT = 3;
    private static final int ACTIVITY_POS = 0;
    private static final int PROJECTS_POS = 1;
    private static final int MEMBERS_POS = 2;

    private final Group group;
    private final String[] titles;
    private final Set<Integer> disabledSections = new HashSet<>();

    public GroupPagerAdapter(Context context, FragmentManager fm, Group group) {
        super(fm);

        this.group = group;
        titles = context.getResources().getStringArray(R.array.group_tabs);
    }

    @Override
    public int getCount() {
        return SECTION_COUNT - disabledSections.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        position = getCorrectPosition(position);

        return titles[position];
    }

    @Override
    public Fragment getItem(int position) {
        position = getCorrectPosition(position);

        switch (position) {
            case ACTIVITY_POS:
                return FeedFragment.newInstance(group.getFeedUrl());
            case PROJECTS_POS:
                return ProjectsFragment.newInstance(group);
            case MEMBERS_POS:
                return GroupMembersFragment.newInstance(group);
        }

        throw new IllegalStateException("Position exceeded on view pager");
    }

    private int getCorrectPosition(int position) {
        for (int i = 0; i <= position; i++) {
            if (disabledSections.contains(i)) {
                position++;
            }
        }

        return position;
    }
}

File: app/src/main/java/com/commit451/gitlab/adapter/GroupPagerAdapter.kt
package com.commit451.gitlab.adapter

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.commit451.gitlab.R
import com.commit451.gitlab.extension.feedUrl
import com.commit451.gitlab.fragment.FeedFragment
import com.commit451.gitlab.fragment.GroupMembersFragment
import com.commit451.gitlab.fragment.ProjectsFragment
import com.commit451.gitlab.model.api.Group
import java.util.*

/**
 * Group pager adapter
 */
class GroupPagerAdapter(context: Context, fm: FragmentManager, private val group: Group) : FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    companion object {

        private const val SECTION_COUNT = 3
        private const val ACTIVITY_POS = 0
        private const val PROJECTS_POS = 1
        private const val MEMBERS_POS = 2
    }

    private val titles: Array<String> = context.resources.getStringArray(R.array.group_tabs)
    private val disabledSections = HashSet<Int>()

    override fun getItem(position: Int): Fragment {
        when (getCorrectPosition(position)) {
            ACTIVITY_POS -> return FeedFragment.newInstance(group.feedUrl)
            PROJECTS_POS -> return ProjectsFragment.newInstance(group)
            MEMBERS_POS -> return GroupMembersFragment.newInstance(group)
        }

        throw IllegalStateException("Position exceeded on view pager")
    }

    override fun getCount(): Int {
        return SECTION_COUNT - disabledSections.size
    }

    override fun getPageTitle(position: Int): CharSequence {
        return titles[getCorrectPosition(position)]
    }

    private fun getCorrectPosition(position: Int): Int {
        var correctPosition = position
        for (i in 0..position) {
            if (disabledSections.contains(i)) {
                correctPosition++
            }
        }

        return correctPosition
    }
}
